config/include.go

```diff
@@ -470,3 +501,203 @@ func mergeHooks(terragruntOptions *options.TerragruntOptions, childHooks []Hook,
+// function.
+func updateBareIncludeBlockJSON(fileBytes []byte) ([]byte, bool, error) {
+	var parsed map[string]interface{}
+	if err := json.Unmarshal(fileBytes, &parsed); err != nil {
+		return nil, false, errors.WithStackTrace(err)
+	}
+	includeBlock, hasKey := parsed["include"]
+	if !hasKey {
+		// No include block, so don't do anything
+		return fileBytes, false, nil
+	}
+	switch typed := includeBlock.(type) {
+	case []interface{}:
+		if len(typed) == 0 {
+			// No include block, so don't do anything
+			return nil, false, nil
+		} else if len(typed) > 1 {
+			// Could be multiple bare includes, or Case 3. We simplify the handling of this case by erroring out,
+			// ignoring the possibility of Case 3, which, while valid HCL encoding, is too complex to detect and handle
+			// here. Instead we will recommend users use the object encoding.
+			return nil, false, errors.WithStackTrace(MultipleBareIncludeBlocksErr{})
+		}
+
+		// Make sure this is Case 2, and not Case 3 with a single labeled block. If Case 2, update to inject the labeled
+		// version. Otherwise, return without modifying.
+		singleBlock := typed[0]
+		if jsonIsIncludeBlock(singleBlock) {
+			return updateSingleBareIncludeInParsedJSON(parsed, singleBlock)
+		}
+		return nil, false, nil
+	case map[string]interface{}:
+		if len(typed) == 0 {
+			// No include block, so don't do anything
+			return nil, false, nil
+		}
+
+		// We will only update the include block if we detect the object to represent an include block. Otherwise, the
+		// blocks are labeled so we can pass forward to the tg parser step.
+		if jsonIsIncludeBlock(typed) {
+			return updateSingleBareIncludeInParsedJSON(parsed, typed)
+		}
+		return nil, false, nil
+	}
+
+	return nil, false, errors.WithStackTrace(IncludeIsNotABlockErr{parsed: includeBlock})
+}
+
+// updateBareIncludeInParsedJSON replaces the include attribute into a block with the label "" in the json. Note that we
+// can directly assign to the map with the single "" key without worrying about the possibility of other include blocks
+// since we will only call this function if there is only one include block, and that is a bare block with no labels.
+func updateSingleBareIncludeInParsedJSON(parsed map[string]interface{}, newVal interface{}) ([]byte, bool, error) {
+	parsed["include"] = map[string]interface{}{"": newVal}
+	updatedBytes, err := json.Marshal(parsed)
+	return updatedBytes, true, errors.WithStackTrace(err)
+}
+
+// jsonIsIncludeBlock checks if the arbitrary json data is the include block. The data is determined to be an include
+// block if:
+// - It is an object
+// - Has the 'path' attribute
+// - The 'path' attribute is a string
+func jsonIsIncludeBlock(jsonData interface{}) bool {
+	typed, isMap := jsonData.(map[string]interface{})
+	if isMap {
+		pathAttr, hasPath := typed["path"]
+		if hasPath {
+			_, pathIsString := pathAttr.(string)
+			return pathIsString
+		}
+		return false
```

Comment:
NIT: not strictly necessary; could fall through.

Reply:
Ah good point. Removed: b29b8cb